clang/lib/Sema/SemaExpr.cpp

```diff
@@ -3382,7 +3382,9 @@ ExprResult Sema::BuildDeclarationNameExpr(
   case Decl::Function: {
     if (unsigned BID = cast<FunctionDecl>(VD)->getBuiltinID()) {
       if (!Context.BuiltinInfo.isDirectlyAddressable(BID)) {
-        type = Context.BuiltinFnTy;
+        type = (BID == Builtin::BI__builtin_counted_by_ref)
+                   ? Context.BuiltinCountedByRefTy
+                   : Context.BuiltinFnTy;
```

Comment:
I feel like we should keep this as-is and only set the type to `BuiltinCountedByRefTy` when we build the call expression. Any use of a builtin outside a call expression is already rejected by existing code, so you don’t have to worry about that if you do it that way.